City-of-Helsinki/berth-reservations-admin
# VEN-611 | Support pagination for tables

## Description :sparkles:
- Implemented Pagination component
- Implemented `usePagination` custom hook to encapsulate the logic
- Implemented a loading state for the Table component

## Issues :bug:
- Sorting the data is still done on the frontend side, to be changed in a follow-up pull request

### Closes :no_good_woman:
[VEN-611](https://helsinkisolutionoffice.atlassian.net/browse/VEN-611): Implement Pagination component and logic

### Related :handshake:

## Testing :alembic:

### Automated tests :gear:️
https://github.com/City-of-Helsinki/berth-reservations-admin/blob/3d641d249301e46004c99038668935f785f13974/src/common/pagination/__tests__/Pagination.test.tsx#L6
https://github.com/City-of-Helsinki/berth-reservations-admin/blob/4fe20451620d23474dcebb68a91000b6a21a92d4/src/common/utils/__tests__/usePagination.test.tsx#L8

### Manual testing :construction_worker_man:
- Got the customers page `Asiakkaat` from the sidebar
- The table should render only 10 rows with a paginator below it to navigate between pages
- Clicking on one page should fetch the data of the page and add a URL parameter 
- Manual reload to the browser should preserve the page
- Using the browser's back and forward buttons should work as expected

## Screenshots :camera_flash:
![image](https://user-images.githubusercontent.com/23040926/81402937-03bc6e00-913b-11ea-8271-fcbc2fe6e4ee.png)

![Screen-Recording-2020-05-08-at-1](https://user-images.githubusercontent.com/23040926/81403241-a248cf00-913b-11ea-9600-8316d6c20b13.gif)

## Additional notes :spiral_notepad:
- Installed a polyfill to support `:focus-visible`
